openreview/profile/process/request_remove_name_decision_process.py

```diff
@@ -256,20 +256,25 @@ def replace_group_members(group, current_member, new_member):
 
         for related_profile in related_profiles:
             print('Related profile', related_profile.id)
-            relations = [{ **r, "username": preferred_id, "name": preferred_name } for r in related_profile.content['relations'] if username == r.get('username')]
-            print('Relations', relations)
-            client.post_profile(openreview.Profile(
-                referent = related_profile.id, 
-                invitation = '~/-/invitation',
-                signatures = ['~Super_User1'],
-                content = {},
-                metaContent = {
-                    'relations': { 
-                        'values': relations,
-                        'weights': [1] * len(relations) 
-                    }
-                })
-            )
+            new_relations = []
+            old_relations = []
+            for relation in related_profile.content['relations']:
+                if username == relation.get('username'):
+                    old_relations.append(relation)
+                    new_relations.append({ **relation, "username": preferred_id, "name": preferred_name })
+                if old_relations or new_relations:
+                    client.post_profile(openreview.Profile(
+                        referent = related_profile.id, 
+                        invitation = '~/-/invitation',
+                        signatures = ['~Super_User1'],
+                        content = {},
+                        metaContent = {
+                            'relations': { 
+                                'values': new_relations + old_relations,
+                                'weights': ([1] * len(new_relations)) + ([-1] * len(old_relations))
```

Comment:
I am trying to understand this: won't len(new_relations) always be equal to len(old_relations) here, making weights 0? is this right? Am I missing something?

Reply:
new_retation has a different username, the one we are replacing.

For example: you want to delete the name "Ana Martinez" from your profile and my profile has "Ana Martinez" as a co-worker.

When we delete Ana Martinez, we need to delete the relation "Ana Martinez" and add the one one "Celeste Martinez" in my profile.

Reply:
yeah that makes sense, but then wouldn't the weight just be 0 always? Since we are always appending to both lists at the same time, their lenghts will be the same, right?

Since len(new_relations) == len(old_relations)
`'weights': ([1] * len(new_relations)) + ([-1] * len(old_relations))` == 0

Just making sure I understand!

Reply:
Let me add a test that verifies that weight arrays are built correctly.